components/vocabularies/omim/api/src/main/java/org/phenotips/vocabulary/internal/solr/MendelianInheritanceInMan.java

```diff
@@ -243,4 +338,183 @@ private int clear()
         }
         return 1;
     }
+
+    private void index()
+    {
+        try {
+            parseOmimData();
+            loadGenes();
+            loadSymptoms(true);
+            loadSymptoms(false);
+            loadGeneReviews();
+            loadVersion();
+        } catch (NullPointerException ex) {
+            this.logger.error("Failed to prepare the OMIM index: {}", ex.getMessage(), ex);
+        }
+    }
+
+    private void parseOmimData()
+    {
+        try {
+            Reader in =
+                new InputStreamReader(new URL(OMIM_SOURCE_URL).openConnection().getInputStream(),
+                    Charset.forName(ENCODING));
+            for (CSVRecord row : CSVFormat.TDF.parse(in)) {
+                // ignore the comments, moved or removed entries
+                if (row.get(0).startsWith("#") || ("Caret").equals(row.get(0))
+                    || ("REMOVED FROM DATABASE").equals(row.get(2))) {
+                    continue;
+                }
+
+                this.crtTerm = new SolrInputDocument();
+                // set id
+                setCrtTerm(ID_FIELD, row.get(1));
+
+                // set symbol
+                setCrtTerm(SYMBOL_FIELD, SYMBOLS.get(row.get(0)));
+                // set type (multivalued)
+                for (String type : TYPES.get(row.get(0))) {
+                    setCrtTerm(TYPE_FIELD, type);
+                }
+                // set name
+                String name = StringUtils.substringBefore(row.get(2), TITLE_SEPARATOR).trim();
+                setCrtTerm(NAME_FIELD, name);
+                // set short name
+                String shortNameString = StringUtils.substringAfter(row.get(2), TITLE_SEPARATOR).trim();
+                String[] shortNames = StringUtils.split(shortNameString, TITLE_SEPARATOR);
+                for (String shortName : shortNames) {
+                    setCrtTerm(SHORT_NAME_FIELD, shortName.trim());
+                }
+
+                // set synonyms
+                if (StringUtils.isNotBlank(row.get(3))) {
+                    String[] synonyms = StringUtils.split(row.get(3), SYNONYM_SEPARATOR);
```

Comment:
The checkstyle error is easy to fix if these two if-for branches would be replaced by a `setListFieldValue` method.